### DISCUSSION TITLE
From MiXCR v3.0.13 to 4.2.0

### DISCUSSION THREAD

**Discussion by paulhtyang** [milaboratory/mixcr#1012] [Q&A]
To Whom It may Concern,

I run TCR RNA sequencing (UMI) from the ArcherDx technology. The goal is to assemble the full length of TCR region. It was working correctly in the old version (v3.0.13) of MIXCR with the following command.
===========================
mixcr analyze amplicon \
    -s hsa \
   --starting-material rna \
   --5-end  v-primers --3-end j-primers \
   --contig-assembly \
   --adapters no-adapters \
  ./raw/1014_BMA_S14_R1_001.molbar.trimmed.deduped.fastq.gz \
  ./raw/1014_BMA_S14_R2_001.molbar.trimmed.deduped.fastq.gz \
  1014_BMA_mixcr
========================================

However, this command is not working while using the current version of MIXCR.

I also read the documents on the MIXCR website and its discussion forum in order to modify the commands used in the current version of MIXCR below **but failed**. Please advice. Thank you so much!
==================================================
mixcr analyze generic-tcr-amplicon-umi \
	--species hsa \
	--rna   --rigid-left-alignment-boundary \
	--floating-right-alignment-boundary C \
	--tag-pattern '^(R1:*)\^(UMI:N{150})' \
	raw/1014_BMA_S14_R1_001.molbar.trimmed.deduped.fastq.gz \
	raw/1014_BMA_S14_R2_001.molbar.trimmed.deduped.fastq.gz \
	results/1014_BMA_S14

mixcr analyze generic-tcr-amplicon \
	--species hsa \
	--rna \
	--floating-left-alignment-boundary \
	--floating-right-alignment-boundary C \
	--assemble-contigs-by VDJRegion \
	./raw/1014_BMA_S14_R1_001.molbar.trimmed.deduped.fastq.gz \
	./raw/1014_BMA_S14_R2_001.molbar.trimmed.deduped.fastq.gz \
	Result_1014_BMA/1014_BMA_S14

mixcr analyze milab-human-tcr-rna-race-full-length \
	--assemble-contigs-by VDJRegion \
	./raw/1014_BMA_S14_R1_001.molbar.trimmed.deduped.fastq.gz \
	./raw/1014_BMA_S14_R2_001.molbar.trimmed.deduped.fastq.gz \
        full_length/1014_BMA
=========================================

**Comment by PoslavskySV** [comment]
Hi, 

could you please share the exact architecture of the library? The correct command will be something like: 

```
mixcr analyze generic-tcr-amplicon-umi \
    --species hsa \
    --rna \
    --tag-pattern <tag-pattern> \
    --rigid-left-alignment-boundary \
    --floating-right-alignment-boundary C \
      input_R1.fastq.gz \
      input_R2.fastq.gz \
      result
```

but to advice you regarding exact parameter values we need to know the library architecture.

Best,
Stan
